extensions/ui/org.eclipse.smarthome.ui.basic/web-src/smarthome.js

```diff
@@ -1732,6 +1747,64 @@
 		var
 			_t = this;
 
+		_t.subscribeRequestURL = "/rest/sitemaps/events/subscribe";
+		_t.reconnectInterval = null;
+		_t.subscribeResponse = null;
+		_t.suppressErrorsState = false;
+
+		function initSubscription(address) {
+			if (featureSupport.eventSource) {
+				ChangeListenerEventsource.call(_t, address);
+			} else {
+				ChangeListenerLongpolling.call(_t);
+			}
+		}
+
+		function connectionRestoredNavigateCallback() {
+			// This will override _t.navigate back to
+			// its normal state
+			_t.startSubscriber(_t.subscribeResponse);
+			_t.subscribeResponse = null;
+		}
+
+		_t.connectionRestored = function(response) {
+			clearInterval(_t.reconnectInterval);
+
+			// Temporarily replace navigation callback
+			_t.navigate = connectionRestoredNavigateCallback;
+
+			// Once navigation is completed, this will be used
+			// to restart SSE subscription
+			_t.subscribeResponse = response;
+
+			smarthome.UI.hideNotification();
+			// Reload current page without affecting the history
+			smarthome.UI.navigate(smarthome.UI.page, false);
+		};
+
+		_t.connectionError = function() {
+			if (_t.suppressErrorsState) {
+				return;
+			}
+
+			var
+				notify = renderTemplate(o.notifyTemplateOffline, {});
```

Comment:
I assume this line break shouldn't be present (the one between `var` and `notify`).